### PR TITLE
KAFKA-17735: release.py must not use home.apache.org

### PR DESCRIPTION
**Pull request by cmccabe**
Previously, Apache Kafka was uploading release candidate (RC) artifacts
to users' home directories on home.apache.org. However, since this
resource has been decommissioned, we need to follow the standard
approach of putting release candidate artifacts into the appropriate
subversion directory, at https://dist.apache.org/repos/dist/dev/kafka/.
(This is distinct from the release directories at
https://dist.apache.org/repos/dist/release/kafka/)

### CODE REVIEW

File: release/release.py [hunk 1]
```diff
@@ -70,10 +70,12 @@
 import gpg
 import notes
 import preferences
-import sftp
```
Comment (jolshan) [inline release/release.py:73]: did we need to change from sftp to svn for this or is it just an additional change?

**Comment by jolshan** [comment]
Have we run the new code to confirm this uploads correctly?

File: release/release.py (hunk 1 above)
Comment (cmccabe) [reply release/release.py:73]: we do not have any place to sftp to any more, now that `home.apache.org` is gone.

**Comment by cmccabe** [comment]
> Have we run the new code to confirm this uploads correctly?

yes. I have tested this. since it's uploading to the devel section of svn a test run is no big deal.

**Comment by jolshan** [comment]
As an aside, we can probably mark https://issues.apache.org/jira/browse/KAFKA-17328 as a duplicate.

**Review by jolshan — APPROVED** [review]

**Pull request merged**